README.md

```diff
@@ -15,7 +15,43 @@ Capacityset implements a set (unordered, non-repeating) that cannot exceed a
 provisioned size.
 
 ## Install
-This repo is a package.  There is no installation.
+This repo is a package. To install it, just run the `go get` command from your command line:
+```sh
+go get -u github.com/xmidt-org/capacityset
+```
+
+## Examples
+
+Creating a Set and adding values is very easy. You can not add duplicates to the set:
+```go
+s := capacityset.NewCapacitySet(3)
+// the Add method returns a boolean indicating if the operation was successful
+s.Add("banana") // true
+s.Add("peach") // true
+fmt.Println(s.Add("banana")) // prints false because the value is already present in the Set
+```
+
+In order to retrieve values from the Set, you can use the `Pop` function. In addition to that, the `Size` function helps 
+you to keep track of the current size of the Set:
+```go
+s := capacityset.NewCapacitySet(3)
+s.Add("banana") // true
+s.Add("peach") // true
+s.Add("apple") // true
+fmt.Println(s.Size()) // prints 3
+// the Pop method returns a random element and removes it from the Set
+fmt.Println(s.Pop()) // either banana, peach or apple
+fmt.Println(s.Size()) // prints 2
+```
+
+If you try to add an element to a Set and exceed the Set's capacity, the `Add` function will panic:
+```go
+s := capacityset.NewCapacitySet(3)
+s.Add("banana") // true
+s.Add("peach") // true
+s.Add("apple") // true
+s.Add("coconut") // would panic because it exceeds the Set's capacity
+```
```

Comment:
When the capacityset is set to 3 and the fourth item is added it is a blocking call and the above code will not finish, until another thread pops an item out of the set.